go/vt/vtgate/planbuilder/operators/horizon_planning.go

```diff
@@ -167,6 +184,39 @@ func tryPushingDownFilter(ctx *plancontext.PlanningContext, in *Filter) (ops.Ope
 	return rewrite.Swap(in, proj, "push filter under projection")
 }
 
+func tryPushingDownDistinct(in *Distinct) (ops.Operator, *rewrite.ApplyResult, error) {
+	if in.Pushed {
+		return in, rewrite.SameTree, nil
+	}
+	switch src := in.Source.(type) {
+	case *Route:
+		if src.IsSingleShard() {
+			return rewrite.Swap(in, src, "push distinct under route")
+		}
+	case *Distinct:
+		return src, rewrite.NewTree("removed double distinct", src), nil
+	case *Aggregator:
+		return nil, nil, errHorizonNotPlanned()
```

Comment:
this support still needs to be added for distinct on top of the aggregator.
will this be added as part of this PR?